### PR TITLE
feat(@sit-onyx/headless): implement global event listener

### PR DESCRIPTION
**Pull request by JoCa96**
Some files/functions were added in preparation for #1460
Relates to #1518

* Rename `*.ct.ts` ts helper files to fix playwright import issue 
* Use `ref` instead of callback
* Update builder type to allow for no parameters and usage of reservered prop `ref`

### CODE REVIEW

File: packages/headless/src/composables/helpers/useGlobalListener.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  event: DocumentEventMap[K],
+) => void;
+
+export type UseGlobalEventListenerOptions<K extends DocumentEventType> = {
+  type: K;
+  listener: GlobalListener<K>;
+  disabled?: Ref<boolean>;
+};
+
+const GLOBAL_LISTENERS = reactive(new Map<DocumentEventType, Set<GlobalListener>>());
```
Comment (larsrickert) [inline packages/headless/src/composables/helpers/useGlobalListener.ts:14]: This is global state, right? I am not sure if this behaves correctly when used with SSR (e.g. Nuxt), so it might be shared across all clients

File: packages/headless/src/composables/helpers/useGlobalListener.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+export const useGlobalEventListener = <K extends DocumentEventType>({
+  type,
+  listener,
+  disabled,
+}: UseGlobalEventListenerOptions<K>) => {
+  onBeforeMount(() => {
+    if (!disabled) {
+      addGlobalListener(type, listener);
+      return;
+    }
```
Comment (larsrickert) [inline packages/headless/src/composables/helpers/useGlobalListener.ts:61]: Isn't this already covered by the watch below?

File: packages/headless/src/composables/tooltip/createTooltip.ts [hunk 3]
```diff
@@ -106,6 +107,9 @@ export const createTooltip = createBuilder((options: CreateTooltipOptions) => {
 
   return {
     elements: {
+      root: {
```
Comment (larsrickert) [inline packages/headless/src/composables/tooltip/createTooltip.ts:110]: Technically this is a breaking change right? Because if you used this composable, you need to change your code to bind the root element which was not needed before

File: packages/headless/src/utils/vitest.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+export const mockVueLifecycle = () => {
+  const { callbacks } = vi.hoisted(() => ({
+    callbacks: {
+      onBeforeUnmountedCb: null as Callback | null,
+      onUnmountedCb: null as Callback | null,
+    },
+  }));
+
+  vi.mock("vue", async (original) => ({
+    ...((await original()) as object),
```
Comment (larsrickert) [inline packages/headless/src/utils/vitest.ts:25]:
```suggestion
    ...((await original()) as typeof import("vue")),
```

File: packages/headless/src/utils/vitest.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    onBeforeMount: (cb: Callback) => cb(),
```
Comment (larsrickert) [inline packages/headless/src/utils/vitest.ts:26]:
```suggestion
    onBeforeMount: vi.fn().mockImplementation((cb: Callback) => cb())
```

Should we use vi.fn() here so they can be spyed on?

File: packages/sit-onyx/playwright.config.ts [hunk 6]
```diff
@@ -40,9 +40,5 @@ export default defineConfig({
     },
   },
   /* Configure projects for major browsers */
-  projects: [
-    { name: "chromium", use: { ...devices["Desktop Chrome"] } },
-    { name: "firefox", use: { ...devices["Desktop Firefox"] } },
-    { name: "webkit", use: { ...devices["Desktop Safari"] } },
-  ],
+  projects: [{ name: "chromium", use: { ...devices["Desktop Chrome"] } }],
```
Comment (larsrickert) [inline packages/sit-onyx/playwright.config.ts:43]: Why did you remove firefox and webkit? Is this unintentionally?

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/playwright.config.ts (hunk 6 above)
Comment (JoCa96) [reply packages/sit-onyx/playwright.config.ts:43]: Yes, whoops 🙈

File: packages/headless/src/composables/helpers/useGlobalListener.ts (hunk 2 above)
Comment (JoCa96) [reply packages/headless/src/composables/helpers/useGlobalListener.ts]: Actually, `watchEffect` seems like a good choice here:

File: packages/headless/src/composables/helpers/useGlobalListener.ts (hunk 1 above)
Comment (JoCa96) [reply packages/headless/src/composables/helpers/useGlobalListener.ts:14]: Yes, it's global state. But listeners are only added on the client onBeforeMount, so everything should be fine in that regard.

File: packages/headless/src/composables/tooltip/createTooltip.ts (hunk 3 above)
Comment (JoCa96) [reply packages/headless/src/composables/tooltip/createTooltip.ts:110]: Updated the Changeset

**Review by larsrickert — APPROVED** [review]
